abci/types/types.go

```diff
@@ -1,125 +1,169 @@
 package types
 
 import (
-	"bytes"
 	"encoding/json"
 
-	"github.com/cosmos/gogoproto/jsonpb"
-)
+	"github.com/cosmos/gogoproto/grpc"
 
-const (
-	CodeTypeOK uint32 = 0
+	v1 "github.com/cometbft/cometbft/api/cometbft/abci/v1"
+	v2 "github.com/cometbft/cometbft/api/cometbft/abci/v2"
+	v3 "github.com/cometbft/cometbft/api/cometbft/abci/v3"
 )
 
-// IsOK returns true if Code is OK.
-func (r ResponseCheckTx) IsOK() bool {
-	return r.Code == CodeTypeOK
-}
-
-// IsErr returns true if Code is something other than OK.
-func (r ResponseCheckTx) IsErr() bool {
-	return r.Code != CodeTypeOK
-}
-
-// IsOK returns true if Code is OK.
-func (r ExecTxResult) IsOK() bool {
-	return r.Code == CodeTypeOK
-}
-
-// IsErr returns true if Code is something other than OK.
-func (r ExecTxResult) IsErr() bool {
-	return r.Code != CodeTypeOK
-}
-
-// IsOK returns true if Code is OK.
-func (r ResponseQuery) IsOK() bool {
-	return r.Code == CodeTypeOK
-}
-
-// IsErr returns true if Code is something other than OK.
-func (r ResponseQuery) IsErr() bool {
-	return r.Code != CodeTypeOK
-}
-
-// IsAccepted returns true if Code is ACCEPT
-func (r ResponseProcessProposal) IsAccepted() bool {
-	return r.Status == ResponseProcessProposal_ACCEPT
-}
-
-// IsStatusUnknown returns true if Code is UNKNOWN
-func (r ResponseProcessProposal) IsStatusUnknown() bool {
-	return r.Status == ResponseProcessProposal_UNKNOWN
-}
-
-func (r ResponseVerifyVoteExtension) IsAccepted() bool {
-	return r.Status == ResponseVerifyVoteExtension_ACCEPT
-}
+type Request = v3.Request
+type RequestEcho = v1.RequestEcho
+type RequestFlush = v1.RequestFlush
+type RequestInfo = v2.RequestInfo
+type RequestInitChain = v3.RequestInitChain
+type RequestQuery = v1.RequestQuery
+type RequestCheckTx = v1.RequestCheckTx
+type RequestCommit = v1.RequestCommit
+type RequestListSnapshots = v1.RequestListSnapshots
+type RequestOfferSnapshot = v1.RequestOfferSnapshot
+type RequestLoadSnapshotChunk = v1.RequestLoadSnapshotChunk
+type RequestApplySnapshotChunk = v1.RequestApplySnapshotChunk
+type RequestPrepareProposal = v3.RequestPrepareProposal
+type RequestProcessProposal = v3.RequestProcessProposal
+type RequestExtendVote = v3.RequestExtendVote
+type RequestVerifyVoteExtension = v3.RequestVerifyVoteExtension
+type RequestFinalizeBlock = v3.RequestFinalizeBlock
+
+// Discriminated Request variants are defined in the latest proto package.
+type Request_Echo = v3.Request_Echo
```

Comment:
To resolve the linter errors, I'd recommend either doing a `//nolint:revive` right at the top of the file to cover the whole file, or to update the [linter settings](https://github.com/cometbft/cometbft/blob/main/.golangci.yml) to ignore this file. I think turning off the specific lint is probably best.